src/Storage/Bucket.php

```diff
@@ -671,4 +739,28 @@ public function name()
     {
         return $this->identity['bucket'];
     }
+
+    /**
+     * Returns whether the bucket with the given file prefix is writable.
+     * Tries to create a temporary file as a resumable upload which will
+     * not be completed (and cleaned up by GCS).
+     *
+     * @param  string $file Optional file to try to write.
+     * @return boolean
+     */
+    public function isWritable($file = null)
```

Comment:
I did a little looking around and couldn't find anything either.  It does appear, though, that an attempt to write without the proper permissions should return a 403 error code. What do you think about checking to make sure the exception is a 403 before returning false? It could help rule out false positives where perhaps 500 level errors may be getting triggered.